src/battle_intro.c

```diff
@@ -600,129 +600,18 @@ void sub_8118FBC(int bgId, u8 arg1, u8 arg2, u8 battlerPosition, u8 arg4, u8 *ar
     LoadBgTilemap(bgId, arg6, BG_SCREEN_SIZE, 0);
 }
 
-#ifdef NONMATCHING
 void unref_sub_8119094(u8 arg0, u8 arg1, u8 battlerPosition, u8 arg3, u8 arg4, u16 arg5, u8 arg6, u8 arg7)
 {
-    int i, j;
-    int offset;
+    int i, j, offset;
+
     DmaCopy16(3, gMonSpritesGfxPtr->sprites[battlerPosition] + BG_SCREEN_SIZE * arg3, (void *)BG_SCREEN_ADDR(0) + arg5, BG_SCREEN_SIZE);
     offset = (arg5 >> 5) - (arg7 << 9);
     for (i = arg1; i < arg1 + 8; i++)
     {
         for (j = arg0; j < arg0 + 8; j++)
         {
-            ((u16 *)BG_VRAM)[i * 32 + j + (arg6 * 0x400) + arg0] = offset | (arg4 << 12);
+            *((u16 *)(BG_VRAM) + (i * 32) + (j + (arg6 << 10))) = offset | (arg4 << 12);
```

Comment:
Wonder if there's a combination like this that will fix that last remaining function in [asm/decompress.s](/pret/pokeemerald/blob/master/asm/decompress.s) 🤔

Reply:
I attempted that one many times, but couldn't get it right, this one is way simpler.